### PR TITLE
(hopefully) fixes the intelligence skillpoint item disappearing

### PR DESCRIPTION
**Pull request by byBackfish**
(hopefully) fixes the intelligence skillpoint item disappearing under certain circumstances.

image
Context: https://canary.discord.com/channels/394189072635133952/946471329857675307/1009132445330845726

### CODE REVIEW

File: common/src/main/java/com/wynntils/wynn/item/IntelligenceSkillPointsItemStack.java [hunk 1]
```diff
@@ -62,6 +62,9 @@ private List<Component> getManaTables() {
         for (int j = 0; j < 4; j++) {
             SpellType spell = SpellType.forClass(characterInfo.getClassType(), j + 1);
 
+            if (spell == null) {
```
Comment (HighCrit) [inline common/src/main/java/com/wynntils/wynn/item/IntelligenceSkillPointsItemStack.java:65]: This does fix the NPE but doesn't fix `spell` being `null`. Which shouldn't be as we have fail over spells with class `None`.

**Review by HighCrit — CHANGES_REQUESTED** [review]

File: common/src/main/java/com/wynntils/wynn/objects/SpellType.java [hunk 2]
```diff
@@ -156,6 +157,7 @@ public static SpellType forClass(ClassType classRequired, int spellNumber) {
                 return spellType;
             }
         }
+        WynntilsMod.error("SpellType not found for " + classRequired + " | Spell: " + spellNumber);
         return null;
```
Comment (HighCrit) [inline common/src/main/java/com/wynntils/wynn/objects/SpellType.java:161]: I.e. we should prevent this being reachable and throw an illegal argument exception if we do somehow reach it.

**Comment by byBackfish** [comment]
Fixed. Turns out the user was using `autojoin` (`/toggle autojoin`) and it therefore skipped the class screen, resulting in artemis not knowing what class he was

**Comment by HighCrit** [comment]
> Fixed. Turns out the user was using `autojoin` (`/toggle autojoin`) and it therefore skipped the class screen, resulting in artemis not knowing what class he was

Still a problem that needs solving 👀

**Comment by kristofbolyai** [comment]
> > Fixed. Turns out the user was using `autojoin` (`/toggle autojoin`) and it therefore skipped the class screen, resulting in artemis not knowing what class he was
> 
> 
> 
> Still a problem that needs solving 👀

It will be in a new PR shortly.

**Pull request closed without merging**